Redesigning Arthropods Part 3: Wasps!

Venera3 opened the pull request:
#### Summary

Content "Wasp rework"

#### Purpose of change

So wasps are cool, right? In contrast to the previous insects they are certainly not forgettable ingame, having firmly established a reputation as an early-midgame menace of all survivors unlucky enough to ignore paper walls. Still, they had no life cycle, no clear place in the difficulty curve and worst of all, no sting. 

#### Describe the solution

- Added wasp eggs, laid by small/giant wasp queens on a 5-day reproduction timer (value pending)
- Eggs hatch into larvae, immobile sacks of mutant flesh with a 16-day growth cycle into
- Pupae, still immobile, grow into giant wasps/ wasp guards / dermatiks (to show brood parasitism)
- Added (giant) wasp foragers, who take a good bit before they decide to hunt you and can be dissuaded with a few good hits
- Added (giant) wasp guards, easily annoyed guardians of the nest that aggro easily and don't run
- Added a queen variant to both the small and giant wasps. The latter is a late-game murder machine, the former is a giant-equivalent that can spawn Day 1 in the nest map extras
- Adjusted the nest map extra to be a bit less of a pushover
- Added some harvest snippets, and adjusted the flying insect harvest lists slightly

TODO:

- [x] Add eggs to the queen's harvest lists
- [x] Adjust wilderness spawns according to the new timeline
- [x] Adjust the map extra's spawn chances - also added them to forests at half the spawn rate of spider nests
- [x] Add the new monsters to the wasp house map extra, remove the rare loot spawn
- [x] Add some foragers to random locations of interest

#### Describe alternatives you've considered

Submit it together with the wasp radio tower location this grew out of, but it's already pretty chunky and I want to go easy on reviewers.
Not add pupae, but they fit and they should only occur in well-protected nests eliminating my usual problem with chrysalis/pupal stages.

#### Testing

Combat testing is mostly done, I'll write up the gist tomorrow.
The new critters spawn, queens lay eggs, eggs grow up. Reproduction fine-tuning follows.

#### Additional context

The timeline for the giants appearing is still in flux, but the general idea is as follows: in the first month, you'll only see the smaller variants of both queens and workers; after a month there's a 50-50 chance of each spawning (and since the small variants evolve into the big ones those will be a bit more common); after two months game-time only giant variants will spawn (and most small ones will have grown into a giant or disqualified from the evolution altogether). The map extras have a 50/50 chance to spawn guards/foragers.

New spawn locations for random foragers (all using the above timeline): 
- Bakeries, ice cream-, butcher-, and candy shops in cities
- Apple orchards, sugar houses, grape fields
- Lumber mills and tree farms

Depends on #47792  for the anger trigger.
Part of #47275 

### Combat balance

They universally have pretty low HP pools, high dodge and can bleed, so cutting/piercing weapons with + to hit are your friends. Their AP coupled with the weakening venom means you *need* to do damage in the first half of the combat (especially early-game) before your effectivity tanks. The foragers have a tendency to flee and bleed out on the way, but I'm still plinking away at their morale a bit.

As usual, I tested 3 character builds, Brawling+TAB for melee, 10 tile range for ranged:
- Day1 (Evacuee gear minus messenger bag, no skills, cudgel/makeshift machete/sharpened rebar/Glock19):
Can fight off a single small forager 8+/10, a small guard 3+/10 times (gun>machete>rebar>cudgel), splats against 3 small or a single giant
- Midgame (Street King gear, 3/3/3 skills, ironshod quarterstaff/broadsword/homemade halfpike/M4 ):
Kills a single small forager/guard 10/10 (same gun>cut>pierce>blunt ranking), 3 small guards 3+/10 and a single giant guard 4+/10 times, taking care of more giants requires a gun and is anything but consistent, splats against a giant queen
- Lategame (Survivor armor+ESAPI vest, 7/7/7 skills lucerne hammer/nodachi/pike, ranged TBD):
Nodachi user kills both a single and 3 giant guards 10/10 times, Queen 5/10 times, Lucerne hammer kills a single giant guard 10/10 times but doesn't have the DPS to deal with 3 or a queen, Pierce user kills both 3 guards and a Queen 4/10 times.

All this should discourage bothering a nest without a long-range weapon, with a lategame melee character *just* being able to clear one by the skin of their teeth.

actual-nh commented:
A venom most effective at causing pain is about right for social wasps; I will give links to some interesting (generally PMC) articles if desired.

Saicchi commented:
I've got an idea for the Dermatik wasp effect which also applies to fungal infection.
What if it the larvae directly consumed the player calories? At a rate of  23 ~ 28 kcal per turn, such that by the time the larvae spawn from you, they consumed a total of 2 kkcal ~ 2.5 kkcal.

Venera3 commented:
> A venom most effective at causing pain is about right for social wasps, particularly the guardian types; I will give links to some interesting (generally PMC) articles if desired.

I considered adding pain for their venom back when I did those, but it makes the stat debuff scaling too messy and runs into the limit that each effect can only have one level of resist trait. Also it would have just made it a straight-up better venom than the damaging kind.

Though with a queen's sting doing up to 45 damage I might want to reword the application message from "the sting itself barely hurt"..

> Dermatik calories

Yes, that would make sense. The problem I have is effects can't yet influence calories and I'd like to keep it as an effect to allow it to be applied to mobs. There's a chunky frozen PR that adds that functionality, so I think I'll wait with them until that merges.

actual-nh commented:
> > A venom most effective at causing pain is about right for social wasps, particularly the guardian types; I will give links to some interesting (generally PMC) articles if desired.
> 
> I considered adding pain for their venom back when I did those, but it makes the stat debuff scaling too messy and runs into the limit that each effect can only have one level of resist trait. Also it would have just made it a straight-up better venom than the damaging kind.

What other resist trait would there be? (The anti-pain ones, as far as I can tell, directly halt the pain, halt it causing problems, or up painkiller.)

Venera3 commented:
Yes, the anti-pain / deadened line, but I didn't check how those work under the hood. The problem would persist that the venom and the pain debuffs would overlap pretty heavily. When we dig into the wound/blood system there'll be room for differentiation (with wasp/bee venom causing distributive shock and pain being more of a distracting factor).

souricelle commented:
Should the smallest wasps really be doing 2d4+8 cut? That's pretty massive damage/bleed for something the size of a cat, and there's poison on top of that. That would be like a single IRL wasp killing a rabbit by bleeding it out.

Venera3 commented:
> 
> 
> Should the smallest wasps really be doing 2d4+8 cut? That's pretty massive cut damage/bleed for something the size of a cat. Currently they do the same damage and the way they kill you is by bleeding you out extremely quickly which seems wrong, the poison barely factors in.

They shouldn't, so they won't. If you read the diff they got 2d3 base damage, have a lower base speed and an attack cost to match, so they only bash you once a round. Most of their damage output will come from their sting and their bites, as it should be.

souricelle commented:
Oops. I've been staring at jsons too long, my eyes are crossing.

actual-nh commented:
#47792 is in on 0.F-dev, which you may wish to rebase this on - or ZhilkinSerg can do it if asked. (BTW, #48077 re factions has been adjusted; could you take another look? I'm hoping to make it not WIP soon, especially with #48069  merged into master.)

Venera3 commented:
Sorry for pinging most of the dev team for a review, rebasing onto 0.F dev but still wanting to merge into master was not a good idea apparently (and I didn't even get the commit I needed anyway).

actual-nh commented:
> Sorry for pinging most of the dev team for a review, rebasing onto 0.F dev but still wanting to merge into master was not a good idea apparently (and I didn't even get the commit I needed anyway).

I'm sorry, the only ones I've done were originally targeted on 0.F-dev; I don't know how @ZhilkinSerg (and others) also switch the targeted branch if it involves more than rebasing (apparently so).

Venera3 commented:
Adjusted the scaling slightly in the default mongroups - there's a larger overlap between the small and the giant versions, and each get a larger packsize to keep up later on:
After a month, small variants can spawn in packs of 2-4, along with the occasional solo giant
After three months, small variants won't spawn at all, and will be replaced by swarming giants.

Also added a one-in-five chance to spawn dermatiks, ~~hopefully getting clang-tidy off my back about not using a tripoint somewhere~~ show off some brood parasitism.

actual-nh commented on data/json/snippets/mutant_anatomy.json:
```diff
@@ -268,7 +268,7 @@
       "The chaotic bundle of organs in its torso look much like those of other insects, but as you near the abdomen there's order to the madness: neat rows of venom-filled glands, each feeding into a reservoir at the base of the stinger",
```
Is this indicating that other insects have chaotic bundles of organs?

Venera3 replied:
Yes, basically the front looks just as crazypants as the insides of the other weird fauna, but where it really matters form follows function.

actual-nh replied:
```suggestion
      "The chaotic bundle of organs in its torso look much like those of other mutated insects, but as you near the abdomen there's order to the madness: comparatively neat rows of venom-filled glands, each feeding into a reservoir at the base of the stinger.",
```
Or "enlarged". "comparatively" could also be rephrased; say, pseudostratified tissue is more organized than a late-stage cancer, but that doesn't mean it's set out as if with a ruler.

Venera3 commented:
There is a very undocumented peculiarity in that mongroup start/end times are multiplied by the set evolution factor, meaning every timer was 4 times as long by default than intended. Fixing all my groups (and wilderness.json) will be a new PR, including the docs and an audit of all other groups for another undocumented bug, but these groups should go live correctly.

kevingranade commented:
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/why-are-wasps-unable-to-reproduce/26342/7

ZhilkinSerg commented on data/json/monstergroups/wilderness.json:
```diff
@@ -2664,26 +2602,35 @@
-        "ends": 180,
+        "pack_size": [ 1, 2 ],
+        "starts": 336,
+        "ends": 720
```
```suggestion
        "ends": 720,
```

EvgenijM86 commented:
I encountered a wasp nest and I have only Wasp Queen generated inside of it. This is on winter start.

![image](https://user-images.githubusercontent.com/430076/125329292-d669e200-e334-11eb-86d4-6840f4e397ce.png)

This is latest git pull at the time of writing this and no extra mods enabled (only what is enabled by default).

Venera3 commented:
Right, they and pupae/larvae don't have a spawn conditional. Thanks, nice catch! The nest should really be empty come winter.
